joplin/base/wagtail_hooks.py

```diff
@@ -300,6 +301,158 @@ def register_link_handler(features):
+                        language=language)
+            new_url.save()
+            new_urls.append(new_url)
+
+        # If we're any other page type, we need to make all our urls
+        # /theme_slug/topic_collection_slug/topic_slug/page_slug/
+        # /department_slug/page_slug/
+        # /page_slug/
+        else:
+            for page_topic in page.topics.all():
+                for topic_page_topic_collection in page_topic.topic.topiccollections.all():
+                    new_url = JanisUrl.create(
+                                information_page=page if page_type == 'InformationPage' else None,
+                                service_page=page if page_type == 'ServicePage' else None,
+                                guide_page=page if page_type == 'GuidePage' else None,
+                                official_documents_page=page if page_type == 'OfficialDocumentPage' else None,
+                                form_container=page if page_type == 'FormContainer' else None,
+                                topic_page=page_topic.topic,
+                                topic_collection_page=topic_page_topic_collection.topiccollection,
+                                theme=topic_page_topic_collection.topiccollection.theme,
+                                page_type=page_type,
+                                language=language)
+                    new_url.save()
+                    new_urls.append(new_url)
+
+            for page_department in page.related_departments.all():
+                new_url = JanisUrl.create(
+                    information_page=page if page_type == 'InformationPage' else None,
+                    service_page=page if page_type == 'ServicePage' else None,
+                    guide_page=page if page_type == 'GuidePage' else None,
+                    official_documents_page=page if page_type == 'OfficialDocumentPage' else None,
+                    form_container=page if page_type == 'FormContainer' else None,
+                    department_page=page_department.related_department,
+                    page_type=page_type,
+                    language=language)
+                new_url.save()
+                new_urls.append(new_url)
+
+            if page.coa_global:
+                new_url = JanisUrl.create(
+                    information_page=page if page_type == 'InformationPage' else None,
+                    service_page=page if page_type == 'ServicePage' else None,
+                    guide_page=page if page_type == 'GuidePage' else None,
+                    official_documents_page=page if page_type == 'OfficialDocumentPage' else None,
+                    form_container=page if page_type == 'FormContainer' else None,
+                    page_type=page_type,
+                    language=language)
+                new_url.save()
+                new_urls.append(new_url)
+
+
+
+    # todo: This can definitely be less copypasta'd and more pythonic
```

Comment:
I agree ;-) <3

Reply:
any suggestions?

Reply:
I'd want to sit down with it and spend some time with it